tests/unistd/open.c

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2021 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
+int main() {
+  int fd = open("dev/stdout/", O_RDONLY);
+
+  dprintf(fd, "WORKING\n");
+
+  // Close open file
+  close(fd);
+
+  // Attempt to write to a non-existent fd.
+  errno = 0;
+  dprintf(fd, "SHOULD NOT PRINT\n");
+  assert(errno == EBADF);
+  printf("Errno: %s\n", strerror(errno));
+
+  // Attempt to open and then read/write to a directory
+  int fd2 = open("dev", O_RDONLY);
+
+  const char* msg = "Test\n";
+
+  errno = 0;
+  int result = write(fd2, msg, strlen(msg));
+  assert(errno == EISDIR);
+  printf("Errno: %s\n", strerror(errno));
+
+  char buf[100];
+
+  errno = 0;
+  int bytes = read(fd2, buf, sizeof(buf));
+  assert(errno == EISDIR);
+  printf("Errno: %s\n", strerror(errno));
```

Comment:
Why add a new test here?   Don't we have enough coverage of the open syscall in our existing tests?  Also, it seems odd to also be testing read and write if the test is called `open.c`.  Is this test designed to be temporary perhaps?